tests/dragonfly/snapshot_test.py

```diff
@@ -500,3 +502,33 @@ async def test_tiered_entries_throttle(async_client: aioredis.Redis):
 
     await load_task
     assert await StaticSeeder.capture(async_client) == start_capture
+
+
+@dfly_args({"proactor_threads": 1, "dbfilename": "big_snapshot"})
+async def test_big_value_serialization_memory_limit(df_factory):
+    instance = df_factory.create()
+    instance.start()
+    client = instance.client()
+
+    ten_mb = 125000
+
+    def ten_mb_random_string():
+        return "".join(random.choices(string.ascii_letters, k=ten_mb))
+
+    one_gb = 125000000
+    upper_limit = one_gb * 1.25
```

Comment:
I'd take the current RSS value, and add (say) 100mb to it as a threshold.
This way the test will be more robust to unrelated, general changes in Dragonfly's memory